srf_generation/source_parameter_generation/gcmt_to_srfgen.py

```diff
@@ -0,0 +1,206 @@
+from srf_generation.source_parameter_generation.uncertainties.common import (
+    GCMT_PARAM_NAMES,
+    GCMT_Source,
+)
+from srf_generation.source_parameter_generation.uncertainties.versions import (
+    load_perturbation_function,
+)
+
+
+def load_args(primary_logger: Logger):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("version", type=str)
+    parser.add_argument("fault_selection_file")
+    parser.add_argument("gcmt_file")
+    parser.add_argument("-n", "--n_processes", default=1)
+    parser.add_argument("-c", "--cybershake_root", type=str, default=".")
+    parser.add_argument("-a", "--aggregate_file", type=str)
+
+    args = parser.parse_args()
+
+    args.fault_selection_file = abspath(args.fault_selection_file)
+    args.gcmt_file = abspath(args.gcmt_file)
+    args.cybershake_root = abspath(args.cybershake_root)
+    if args.aggregate_file is not None:
+        args.aggregate_file = abspath(args.aggregate_file)
+
+    errors = []
+
+    if not isfile(args.fault_selection_file):
+        errors.append(f"Specified station file not found: {args.fault_selection_file}")
+    if not isfile(args.gcmt_file):
+        errors.append(f"Specified gcmt file not found: {args.gcmt_file}")
+
+    if errors:
+        message = (
+            "At least one error was detected when verifying arguments:\n"
+            + "\n".join(errors)
+        )
+        raise ValueError(message)
+
+    makedirs(args.cybershake_root, exist_ok=True)
+
+    log_file = join(args.cybershake_root, primary_logger.name + "_log.txt")
+    add_general_file_handler(primary_logger, log_file)
+
+    primary_logger.debug(f"Arguments parsed: {args}")
+
+    return args
+
+
+def generate_uncertainties(
+    data: GCMT_Source,
+    realisation_count: int,
+    cybershake_root: str,
+    perturbation_function: Callable,
+    aggregate_file: Union[str, None],
+    primary_logger_name: str,
+):
+    primary_logger = get_logger(name=primary_logger_name)
+    fault_logger = get_realisation_logger(primary_logger, data.pid)
+    fault_logger.debug(f"Fault {data.pid} had data {data}")
+    for i in range(realisation_count):
+        fault_logger.debug(
+            f"Generating realisation {i} of {realisation_count} for fault {data.pid}"
+        )
+        rel_name = get_realisation_name(data.pid, i)
+
+        fault_logger.debug("Calling perturbation function.")
+        perturbed_realisation = perturbation_function(data)
+        fault_logger.debug(
+            f"Got results from perturbation_function: {perturbed_realisation}"
+        )
+
+        file_name = get_srf_path(cybershake_root, rel_name).replace(".srf", ".yaml")
+        makedirs(dirname(file_name), exist_ok=True)
+        fault_logger.debug(
+            f"Created Srf directory and attempting to save perturbated source generation parameters there: {file_name}"
+        )
+        with open(file_name, "w") as params_file:
+            yaml.dump(perturbed_realisation, params_file)
```

Comment:
Is there any case where we have different parameters across realisations? Otherwise, a csv is probably actually better. To avoid storing parameter names 12,000 times...

Reply:
A csv still has a header line too though? :P